tests/subsys/net/lib/lwm2m/lwm2m_registry/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(lwm2m_client_unittest)
+
+FILE(GLOB app_sources src/*.c)
+target_sources(app
+  PRIVATE
+    ${app_sources}
+    $ENV{ZEPHYR_BASE}/subsys/net/lib/lwm2m/ipso_temp_sensor.c
```

Comment:
Ditto, this should be pulled in by the build system (CONFIG_LWM2M_IPSO_TEMP_SENSOR is enabled)